lighthouse-core/report/v2/report-renderer.js

```diff
@@ -25,19 +57,184 @@
  */
 window.ReportRenderer = class ReportRenderer {
   /**
-   * @param {!Object} report
+   * @param {Document} document
+   */
+  constructor(document) {
+    this._document = document;
+  }
+
+  /**
+   * @param {ReportJSON} report
+   * @return {Element}
+   */
+  renderReport(report) {
+    try {
+      const element = this._renderReport(report);
+      return element;
+    } catch (e) {
+      return this._renderException(e);
+    }
+  }
+
+  /**
+   * @param {string} name
+   * @param {string=} className
+   * @param {Array<Object>=} attrs Attribute key/val pairs.
+   * @returns {Element}
+   */
+  _createElement(name, className, attrs = []) {
+    const element = this._document.createElement(name);
+    if (className) {
+      element.className = className;
+    }
+    for (const [key, val] of Object.entries(attrs)) {
```

Comment:
huh? This is really hard to parse, and I can't find usage of it to help :) What's wrong with `attrs.forEach(attr => el.setAttribute(attr.key, attr.value))`?